python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,457 @@
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+
+        await self.configure_tcs()
+        await self.configure_camera()
+
+        if hasattr(config, "ignore"):
+            for comp in config.ignore:
+                if comp in self.camera.components_attr:
+                    self.log.debug(f"Ignoring Camera component {comp}.")
+                    setattr(self.camera.check, comp, False)
+                else:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Group. "
+                        f"Must be one of {self.camera.components_attr}. "
+                        f"Ignoring."
+                    )
+
+        self.config = config
+
+        await super().configure(config)
+
+    def set_metadata(self, metadata: salobj.BaseMsgType) -> None:
+        """Set script metadata, including estimated duration."""
+
+        n_flats = self.config.n_flat
+
+        # Initialize estimate flat exposure time
+        target_flat_exptime = (self.config.max_exp_time + self.config.min_exp_time) / 2
+
+        # Setup time for the camera (readout and shutter time)
+        setup_time_per_image = self.camera.read_out_time + self.camera.shutter_time
+
+        # Total duration calculation
+        total_duration = (
+            self.instrument_setup_time  # Initial setup time for the instrument
+            + target_flat_exptime * (n_flats)  # Time for taking all flats
+            + setup_time_per_image * (n_flats)  # Setup time p/image
+        )
+
+        metadata.duration = total_duration
+        metadata.instrument = self.get_instrument_name()
+        metadata.filter = self.get_instrument_filter()
+
+    def get_new_exptime(self, sky_counts, exp_time):
+        """Calculate exposure time for next image.
+
+        Parameters
+        ----------
+        sky_counts : float
+            Counts in electrons of previous flat.
+        exp_time : float
+            Exposure time of previous flat
+
+        Returns
+        -------
+        float
+            Calculated new exposure time.
+        """
+
+        return self.config.target_sky_counts * exp_time / sky_counts
+
+    def get_target_radec(self):
+        """
+        Returns the RADEC of the target area of the sky that's an azimuth
+        `distance_from_sun` away from the Sun, given `elevation`,
+        and at a given `time`.
+
+        Parameters
+        ----------
+        distance_from_sun : float, (-180, 180)
+            The distance from the Sun in degrees.
+            Positive angles go towards the North.
+        elevation : float
+            Target elevation for Sky Flats.
+        """
+
+        az_sun, el_sun = self.tcs.get_sun_azel()
+
+        target_az = (az_sun + self.config.distance_from_sun) % 360
+
+        target_radec = self.tcs.radec_from_azel(target_az, self.config.target_el)
+
+        return target_radec
+
+    def get_empty_field(self, target, radius=5):
```

Comment:
You need to convert this method to a coroutine and run the query in an executor (see below). 

Otherwise calling this will block the async event loop and will look like the script has died.